documentation/views.py

```diff
@@ -42,16 +44,20 @@ def embed_view(request, slug, ide_slug):
     return render(request, 'documentation/embed.html', {'code_block': block, 'ide': ide})
 
 
-def page_view(request, slug, curric_slug):
-    try:
-        page = Map.objects.get(slug=slug, parent__slug=curric_slug)
-    except Map.DoesNotExist:
-        raise ContinueResolving
+def page_view(request, parents, slug):
+    pages = Map.objects.filter(slug=slug)
+
+    if pages.count() > 1 and parents is not None:
+        # If more than one map was found with the same slug, try narrowing by parent
+        parent_slug = parents.split('/')[-1]
+        if pages.filter(parent__slug=parent_slug).count() > 0:
+            pages = pages.filter(parent__slug=parent_slug)
```

Comment:
nit: I don't remember exactly how Django caches queries, but I think you'll save a select if you do something like

```python
parent_slug = parents.split('/')[-1]
pages_by_parent = pages.filter(parent__slug=parent_slug)
if pages_by_parent.count() > 0:
  pages = pages_by_parent
```

Reply:
I think the caching middleware caches view queries, but I think it makes sense to go with your proposal anyway